File: reference-solutions/typescript/pacotes-complementares.ts
import { readFileSync } from "node:fs";

// A primeira ocorrência de cada volume é suficiente para desempatar pelo menor i.
// Consultar antes de inserir impede que um pacote forme um par consigo mesmo.
export function solve(input: string): string {
  const values = input.trim().split(/\s+/).map(Number);
  const n = values[0]!;
  const target = values[1]!;
  const firstIndex = new Map<number, number>();
  for (let j = 1; j <= n; j += 1) {
    const volume = values[j + 1]!;
    const i = firstIndex.get(target - volume);
    if (i !== undefined) return `${i} ${j}`;
    if (!firstIndex.has(volume)) firstIndex.set(volume, j);
  }
  return "-1";
}

console.log(solve(readFileSync(0, "utf8")));

File: reference-solutions/python/pacotes-complementares.py
import sys


def solve(data: str) -> str:
    values = list(map(int, data.split()))
    n, target = values[:2]
    first_index: dict[int, int] = {}
    for j in range(1, n + 1):
        volume = values[j + 1]
        i = first_index.get(target - volume)
        if i is not None:
            return f"{i} {j}"
        # Não substituir a primeira ocorrência: ela vence os desempates.
        if volume not in first_index:
            first_index[volume] = j
    return "-1"


if __name__ == "__main__":
    print(solve(sys.stdin.read()))
